CMakeLists.txt

```diff
@@ -0,0 +1,200 @@
+#################################################################################
+# CMakeLists.txt for Mahogany Email Client
+#################################################################################
+# 
+# This is the main CMake configuration file for the Mahogany email client project.
+# It provides a cross-platform build system that mirrors the existing Visual Studio
+# MSBuild configuration while adding support for other platforms.
+#
+# OVERVIEW:
+# ---------
+# Mahogany is a full-featured GUI email client built with wxWidgets that includes:
+# - Main application (M/Mahogany executable)
+# - IMAP library for mail handling
+# - Compface library for X-Face image support
+# - DSPAM library for spam filtering
+# - Versit library for vCard handling
+# - Python scripting interface
+#
+# USAGE:
+# ------
+# 1. Prerequisites:
+#    - CMake 3.16 or later
+#    - C++ compiler with C++11 support
+#    - wxWidgets 3.0 or later
+#    - Python development headers (optional, for Python support)
+#    - OpenSSL (optional, for SSL/TLS support)
+#
+# 2. Basic build:
+#    mkdir build && cd build
+#    cmake ..
+#    make
+#
+# 3. Build with options:
+#    cmake -DBUILD_SHARED_LIBS=ON -DENABLE_PYTHON=ON -DCMAKE_BUILD_TYPE=Release ..
+#    make
+#
+# 4. Install:
+#    make install
+#
+# BUILD OPTIONS:
+# --------------
+# - BUILD_SHARED_LIBS: Build shared libraries instead of static (default: OFF)
+# - ENABLE_PYTHON: Enable Python scripting support (default: ON if Python found)
+# - ENABLE_SSL: Enable SSL/TLS support (default: ON if OpenSSL found)
+# - CMAKE_BUILD_TYPE: Debug, Release, RelWithDebInfo, MinSizeRel
+# - CMAKE_INSTALL_PREFIX: Installation directory
+#
+# TARGETS:
+# --------
+# Libraries:
+# - mahogany_imap: IMAP client library (static/shared)
+# - mahogany_compface: X-Face image library (static/shared)
+# - mahogany_dspam: Spam filtering library (static/shared)
+# - mahogany_versit: vCard library (static/shared)
+#
+# Executables:
+# - mahogany: Main GUI application
+# - mtest: IMAP test utility (if built)
+# - mailutil: Mail utility (if built)
+#
+# PLATFORM SUPPORT:
+# ------------------
+# - Windows (MSVC, MinGW)
+# - Linux (GCC, Clang)
+# - macOS (Clang, GCC)
+# - Other Unix-like systems
+#
+# The CMake configuration automatically detects the platform and applies
+# appropriate compiler flags and library dependencies.
+#
+#################################################################################
+
+cmake_minimum_required(VERSION 3.16)
+
+project(Mahogany 
+    VERSION 0.68.0
```

Comment:
Version should be extracted from `include/Mversion.h` instead of being hardcoded.